render/egl.c

```diff
@@ -246,26 +263,26 @@ struct wlr_egl *wlr_egl_create_with_drm_fd(int drm_fd) {
 			"eglQueryDmaBufModifiersEXT");
 	}
 
-	const char *device_exts_str = NULL, *driver_name = NULL;
-	if (check_egl_ext(client_exts_str, "EGL_EXT_device_query")) {
-		load_egl_proc(&egl->procs.eglQueryDisplayAttribEXT,
-			"eglQueryDisplayAttribEXT");
-		load_egl_proc(&egl->procs.eglQueryDeviceStringEXT,
-			"eglQueryDeviceStringEXT");
+	if (platform == EGL_PLATFORM_DEVICE_EXT && !egl->exts.device_base_ext) {
```

Comment:
If `EGL_PLATFORM_DEVICE_EXT` is not supported, `eglGetPlatformDisplayEXT` should have failed already. We should move this check up to before the `eglGetPlatformDisplayEXT` call.

Reply:
This is the wrong ext to check for `EGL_PLATFORM_DEVICE_EXT` support. The ext that provides the device platform is `EGL_EXT_platform_device`.

Reply:
Made the check against the right ext and moved the platform check at the beginning of `egl_init`